include/libopencm3/sam/4l/pm.h

```diff
@@ -0,0 +1,184 @@
+
+#define PM_PBBSEL			MMIO32(PM_BASE + 0x010)
+/* PBC Clock Select PBCSEL Read/Write */
+
+#define PM_PBCSEL			MMIO32(PM_BASE + 0x014)
+/* PBD Clock Select PBDSEL Read/Write */
+
+#define PM_PBDSEL			MMIO32(PM_BASE + 0x018)
+/* CPU Mask CPUMASK Read/Write */
+
+#define PM_CPUMASK			MMIO32(PM_BASE + 0x020)
+/* HSB Mask HSBMASK Read/Write */
+
+#define PM_HSBMASK			MMIO32(PM_BASE + 0x024)
+/* PBA Mask PBAMASK Read/Write */
+
+#define PM_PBAMASK			MMIO32(PM_BASE + 0x028)
+/* PBB Mask PBBMASK Read/Write */
+
+#define PM_PBBMASK			MMIO32(PM_BASE + 0x02C)
+/* PBC Mask PBCMASK Read/Write */
+
+#define PM_PBCMASK			MMIO32(PM_BASE + 0x030)
+/* PBD Mask PBDMASK Read/Write */
+
+#define PM_PBDMASK			MMIO32(PM_BASE + 0x034)
+/* PBA Divided Mask PBADIVMASK Read/Write */
+
+#define PM_PBADIVMASK			MMIO32(PM_BASE + 0x040)
+/* Clock Failure Detector Control CFDCTRL Read/Write */
+
+#define PM_CFDCTRL			MMIO32(PM_BASE + 0x054)
+/* Unlock Register UNLOCK Write-only */
+
+#define PM_UNLOCK			MMIO32(PM_BASE + 0x058)
+/* Interrupt Enable Register IER Write-only */
+
+#define PM_IER				MMIO32(PM_BASE + 0x0C0)
+/* Interrupt Disable Register IDR Write-only */
+
+#define PM_IDR				MMIO32(PM_BASE + 0x0C4)
+/* Interrupt Mask Register IMR Read-only */
+
+#define PM_IMR				MMIO32(PM_BASE + 0x0C8)
+/* Interrupt Status Register ISR Read-only */
+
+#define PM_ISR				MMIO32(PM_BASE + 0x0CC)
+/* Interrupt Clear Register ICR Write-only */
+
+#define PM_ICR				MMIO32(PM_BASE + 0x0D0)
+
+/* Status Register SR Read-only */
+#define PM_SR				MMIO32(PM_BASE + 0x0D4)
+
+/* Peripheral Power Control Register PPCR Read/Write */
+#define PM_PPCR				MMIO32(PM_BASE + 0x160)
+
+/* Reset Cause Register RCAUSE Read-only */
+#define PM_RCAUSE			MMIO32(PM_BASE + 0x180)
+
+/* Wake Cause Register WCAUSE Read-only */
+#define PM_WCAUSE			MMIO32(PM_BASE + 0x184)
+
+/* Asynchronous Wake Enable AWEN Read/Write */
+#define PM_AWEN				MMIO32(PM_BASE + 0x188)
+
+/* Protection Control Register PROTCTRL Read/Write */
+#define PM_PROTCTRL			MMIO32(PM_BASE + 0x18C)
+
+/* Fast Sleep Register FASTSLEEP Read/Write */
+#define PM_FASTSLEEP			MMIO32(PM_BASE + 0x194)
+
+/* Configuration Register CONFIG Read-only */
+#define PM_CONFIG			MMIO32(PM_BASE + 0x3F8)
+
+/* Version Register VERSION Read-only */
+#define PM_VERSION			MMIO32(PM_BASE + 0x3FC)
+
+
+/* --- Register contents --------------------------------------------------- */
+#define PM_MCCTRL_MCSEL_SHIFT			0
+
+#define PM_CPUSEL_CPUDIV_SHIFT			7
+#define PM_CPUSEL_CPUSEL_SHIFT			0
+
+#define PM_PBSEL_PBDIV_SHIFT			7
+#define PM_PBSEL_PBSEL_SHIFT			0
+
+#define PM_CPUMASK_OSC_SHIFT			0
+
+#define PM_HSBMASK_PDCA_SHIFT			0
+#define PM_HSBMASK_FLASHCALW_SHIFT			1
+#define PM_HSBMASK_FLASHCALW_PICO_SHIFT			2
+#define PM_HSBMASK_USBC_SHIFT			3
+#define PM_HSBMASK_CRCCU_SHIFT			4
+#define PM_HSBMASK_APBA_SHIFT			5
+#define PM_HSBMASK_APBB_SHIFT			6
+#define PM_HSBMASK_APBC_SHIFT			7
+#define PM_HSBMASK_APBD_SHIFT			8
+#define PM_HSBMASK_AESA_SHIFT			9
```

Comment:
If these are all single bits, the convention has been for them to be preshifted, so 
```C
#define PM_HSBMASK_APBC           (1<<7)
```

There's been a bit more back and forth on fields, but single bits have almost always been preshifted.